### PR TITLE
created a more thorough overview for devfiles

### PR DESCRIPTION
**Pull request by jc-berger**
For issue [#159](https://github.com/devfile/api/issues/159)

I've applied the comments and feedback from the previous PR, [#25 ]. Due to doc restructure, it was easiest to close out that PR and branch and start anew. 

@elsony and/or @johnmcollier, tagging either of you to see if this new content is thorough and extensive enough for a solid overview on devfiles for users. 

@rkratky, tagging for your awareness as well.

### CODE REVIEW

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/con_devfile.adoc:6]:
```suggestion
Devfiles are YAML files that describe and define developer workspaces or cloud-native development environments. Devfiles also describe best practices for end-to-end application development while providing a way for being customized or extended to fit a team's current development practices. Instead of forcing a new workflow or tools, devfiles provide instructions for automating what was previously done manually.
```

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/con_devfile.adoc:8]:
```suggestion
Devfiles are consumed by tools that transform the definition into a cloud workspace composed of multiple containers and cloud-native resources. Also, any number of identical workspaces can be created from the same devfile. This way, workspaces can be used by different teams and shared in various ways, including a devfile registry or together with the source code of a project.
```

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (rkratky) [inline docs/modules/user-guide/partials/con_devfile.adoc:14]:
```suggestion
* Customizing workspaces
```

**Review by rkratky — APPROVED** [review]
Just a few nitpicks.

**Comment by jc-berger** [comment]
@rkratky, thanks for the review. I've made a new, squashed commit. 

@elsony and/or @johnmcollier, thorough enough overview do you think?

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (johnmcollier) [inline docs/modules/user-guide/partials/con_devfile.adoc:13]: I don't know if I would phrase `Finding stacks using the devfile registry` this way.

While I think a registry of stacks is certainly a benefit to using devfiles, I think it makes more sense to say something like "A registry for finding and hosting devfiles?"

Comment (jc-berger) [reply docs/modules/user-guide/partials/con_devfile.adoc:13]: @johnmcollier, thanks. I like your rephrase, but for consistency, I want to have every bullet begin with a gerund. What do you think about:
`Having a registry fro finding and hosting devfiles`

Comment (johnmcollier) [reply docs/modules/user-guide/partials/con_devfile.adoc:13]: Yup, I think that  sounds good 👍

**Comment by jc-berger** [comment]
@johnmcollier, thanks! New and squashed commit is up. Would you please comment with `LGTM` if you think it's ready to be merged?

@rkratky, once it has the `LGTM` comment, would you please merge, that is, if it looks good to you too!

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (elsony) [inline docs/modules/user-guide/partials/con_devfile.adoc:17]: Add a link to an example of tools that currently have devfile support, e.g. odo and we'll add dev console later after they have adopted it.

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (elsony) [inline docs/modules/user-guide/partials/con_devfile.adoc:14]: Add (reword if needed):
Extensible to allow easy devfile customization

**Review by elsony — CHANGES_REQUESTED** [review]
A couple of things that I'd like to add to the overview:
1. Includes some concrete examples of things that can be included in the devfile, e.g.
   Includes:
   - Guidance about runtime images
   - Example code
   - Build and CI commands
   - Deployment options
2. Add who can use devfiles (stack providers, tools provides, developers)

**Comment by jc-berger** [comment]
@elsony, thanks for the review. See the new, squashed commit, addressing your feedback. So, I added context to our `additional resources` so users know why they're going to these links. As you suggested, I added who can use devfiles. Let me know what you think about this new section. 

As far as including example code, commands, guidance on runtimes, and deployment options, were you thinking of taking some content from elsewhere in our docs? Going through the devfile docs we have, I didn't see anything that instantly spoke to any of these topics you suggested. So thoughts on how we'll get this content to users on this overview page?

**Comment by elsony** [comment]
> As far as including example code, commands, guidance on runtimes, and deployment options, were you thinking of taking some content from elsewhere in our docs? Going through the devfile docs we have, I didn't see anything that instantly spoke to any of these topics you suggested. So thoughts on how we'll get this content to users on this overview page?

@jc-berger Sorry, maybe I wasn't clear on my initial comment. I only meant to include that list of 4 things (just those 4 entries) as part of the overview docs as the intro.  The user can find the actual details of those from our user guide and schema doc.

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (elsony) [inline docs/modules/user-guide/partials/con_devfile.adoc:16]: Maybe organize it a bit and put these three consumer usage scenario/benefits under a section, e.g. the section name can be something like "Who would use the devfiles?" or "Consumers of devfiles", etc.

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (elsony) [inline docs/modules/user-guide/partials/con_devfile.adoc:24]: Make Additional resources to be a proper section header (e.g. with different font). Currently, it just looks like the rest of the other text.

**Review by elsony — CHANGES_REQUESTED** [review]

**Comment by jc-berger** [comment]
@elsony, thanks for the feedback. I was on vacation when you did your review, hence, the late reply. The new commit is up.

I like your idea of making the sections of the doc a bigger size, comparable to the title's font size. However, throughout the rest of the doc, each section is simply emboldened, not actually a bigger font size. So to be consistent, I left these section titles bold, not making them bigger. As we continue to improve our docs this year, the doc team can discuss how we want to organize our sections! 

Let me know what you think of the changes and if you think it's ready to merge. Thanks!

**Review by elsony — APPROVED** [review]
/lgtm

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (ossontoes) [inline docs/modules/user-guide/partials/con_devfile.adoc:6]: At some RH writing training we were advised to write for "non-native speakers with IELTS level 3" (out of 9). While the meaning of the 2nd sentence seems to be clear, do you think grammar here is on the more advanced side and can cause some readers troubles?

E.g. this part: "while providing a way for being customized or extended to fit". If you decide to rephrase it, possible ways could be "users/you can customize and extend devfiles to..." or "... and can be customized or extended to fit..." (but it's passive voice, it's frowned upon in general).

Comment (ossontoes) [reply docs/modules/user-guide/partials/con_devfile.adoc:6]: Also, this part might also be tricky, WDYT? But not too much, it's up to you, of course, if you'd like to keep it.
   "devfiles provide instructions for automating what was previously done manually"
Will this version be a bit more lightweight?
   "for automating manual processes"

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (ossontoes) [inline docs/modules/user-guide/partials/con_devfile.adoc:8]: If we intended to reduce passive voice usage, possible rephrasing options would be:
- (development?) tools transform the definition...
- Users can create any number of identical workspaces from the same devfile

But as I wrote earlier I don't believe passive voice is a 100% pure evil, but it's a "no-no" in many RH docs teams.

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (ossontoes) [inline docs/modules/user-guide/partials/con_devfile.adoc:13]: *Example?

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (ossontoes) [inline docs/modules/user-guide/partials/con_devfile.adoc:17]: If "include" is removed, will the sentence lose some meaning? Can it be seen as noise? Or maybe you wanted some consistency with the previous list (A single devfile can include)?

File: docs/modules/user-guide/partials/con_devfile.adoc
Comment (ossontoes) [inline docs/modules/user-guide/partials/con_devfile.adoc:40]: It's highly project-specific: in several projects I worked with the common word preceding links was "see" (instead of "go to"), but I've never heard it was a rule or company style. Feel free to ignore this if you use "go to" consistently in the docs.

In several trainings and meetings writers discussed xrefs and the general consensus was "Don't use xrefs" :) However, if you never experienced issues while building docs, I don't see why you shouldn't use them. I guess writers ran into troubles while working on some legacy docs.

**Comment by ossontoes** [comment]
Hi Jake, thank you for a great and well-written text.
I'm done. My suggestions are mostly about simplifying grammar or reducing noise, but then it's too subjective, feel free to ignore me everywhere :)

**Comment by jc-berger** [comment]
@ossontoes, awesome, on point revisions and feedback :) I completely agree about there being too much noise and too complex grammar. 

I've addressed your comments. Please let me know what you think, thanks!

**Comment by jc-berger** [comment]
@elsony, I'm thinking about the links in our "additional resources" section. From a doc perspective, links can be troublesome: they can become outdated, encourage users to quickly leave the page, etc. 

I do like directing users' attention after the overview. But instead of giving several links, maybe only one link to the rest our documentation? What do you think?

If you like the idea of cutting down the "additional resources" section, let me know, and I'll get a new commit so we can see how it'd look in the final version.

**Comment by ossontoes** [comment]
LGTM, thank you @jc-berger

**Comment by elsony** [comment]
@jc-berger How does it look like if there is only one link? Can you provide an example?

**Pull request merged**
